### PR TITLE
s3_sync - add  storage_class feature

### PR DESCRIPTION
**Pull request by alinabuzachis**
##### SUMMARY

s3_sync - add support for chosing storage_class when objects are added to an S3 bucket.
Fixes: https://github.com/ansible-collections/community.aws/issues/358

##### ISSUE TYPE

- Feature Pull Request

### CODE REVIEW

File: tests/integration/targets/s3_sync/tasks/main.yml [hunk 1]
```diff
@@ -55,6 +55,30 @@
...
+    - name: Sync files with remote bucket using glacier storage class
+      s3_sync:
+        bucket: "{{ resource_prefix }}-testbucket-ansible-2"
+        file_root: "{{ output_dir }}/s3_sync"
+        storage_class: "GLACIER"
+      register: output
+
+    - assert:
+        that:
+          - output is changed
```
Comment (tremble) [inline tests/integration/targets/s3_sync/tasks/main.yml:79]: Any way we can test that this is actually pushing to glacier?

**Review by tremble — COMMENTED** [review]
Changelog please :)

File: tests/integration/targets/s3_sync/tasks/main.yml (hunk 1 above)
Comment (alinabuzachis) [reply tests/integration/targets/s3_sync/tasks/main.yml:79]: @tremble I verified in the AWS console. According to the documentation https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/s3.html#S3.Bucket.upload_fileobj, this information doesn't seem to be returned, for this reason I didn't implement a check.

File: changelogs/fragments/497-s3_sync-add-storage_class.yaml [hunk 2]
```diff
@@ -0,0 +1,2 @@
+minor_changes:
+- s3_sync: new storage_class feature allowing to specify the storage_class when any object is added to an s3_bucket (https://github.com/ansible-collections/community.aws/issues/358).
```
Comment (tremble) [inline changelogs/fragments/497-s3_sync-add-storage_class.yaml:2]:
```suggestion
- s3_sync - new ``storage_class`` feature allowing to specify the storage class when any object is added to an S3 bucket (https://github.com/ansible-collections/community.aws/issues/358).
```

**Review by tremble — CHANGES_REQUESTED** [review]
Minor niggle on the changelog, otherwise LGTM

**Review by tremble — APPROVED** [review]
LGTM feel free to merge once the tests pass (or I'll do so this evening)

File: changelogs/fragments/497-s3_sync-add-storage_class.yaml (hunk 2 above)
Comment (alinabuzachis) [reply changelogs/fragments/497-s3_sync-add-storage_class.yaml]: @tremble Fixed. Thank you for reviewing.

**Pull request merged**